src/node_api.h

```diff
@@ -524,6 +524,8 @@ NAPI_EXTERN napi_status napi_get_dataview_info(napi_env env,
 // Methods to manage simple async operations
 NAPI_EXTERN
 napi_status napi_create_async_work(napi_env env,
+                                   napi_value async_resource,
+                                   const char* async_resource_name,
```

Comment:
Can we make the `async_resouce_name` parameter a `napi_value` instead of `char*`? That will resolve the issue in https://github.com/nodejs/node/issues/15108 for this API.

I see `AsyncWrap::EmitAsyncInit()` converts this parameter to a `v8::String` anyway; hopefully we can avoid extra conversion from `napi_value` (which is actually already a `v8::String`) to `char*` and back to `v8::String`.

Reply:
@jasongin We can, and I thought about it. In practice, this is always going to be static (even compile-time constant), so turning this into a `napi_value` would make the API a bit harder to use; I would suggest adding a wrapper if and only if somebody ever asks for that functionality.

If you want to make that change as a matter of principle, that would be okay with me.

Reply:
I agree in C++ code that string is almost always going to be a constant, But the linked issue is about making the APIs possible to consume from other languages (like rust) which may not use null-terminated strings.

I'm not as concerned about usability since most people would use API this through a C++ wrapper, which can automatically create the `napi_value` from the `char*`.

While I might suggest you can leave it this way for now and I can make that change later, that would be two breaking changes instead of one, which is not so good.

Reply:
Also I was wondering if this might prompt some change/addition to the async hooks public API -- should there be an overload that takes a `v8::Local<v8::String>` instead of a `char*`? While we can consume internal APIs from `node_api.cc`, it causes more difficulty when building as an external module.

Reply:
@jasongin Yeah, good points. It’s late here and I will be busy for a lot of the weekend, so feel free to push new commits with those changes into this PR

Reply:
I made this change at https://github.com/jasongin/nodejs/commit/95fa1504b5644f7729e7e03a0d6100491930498c

If you like, you may add that commit onto this PR.

Reply:
@jasongin Done, thank you a lot!